internal/core/services/claims.go

```diff
@@ -597,6 +604,29 @@ func (c *claim) revoke(ctx context.Context, did *w3c.DID, nonce uint64, descript
 	return nil
 }
 
+func (c *claim) getRevocationStatus(ctx context.Context, basicMessage *ports.AgentRequest) (*domain.Agent, error) {
+	revData := &protocol.RevocationStatusRequestMessageBody{}
+	err := json.Unmarshal(basicMessage.Body, revData)
+	if err != nil {
+		return nil, fmt.Errorf("invalid revocation request body: %w", err)
+	}
+
+	var revStatus *verifiable.RevocationStatus
```

Comment:
you should reuse [this
](https://github.com/iden3/go-schema-processor/blob/adbd0a5aa31811583d88aa9f48ff00b4bb1efd9f/verifiable/credential_status.go#L33)

Reply:
@vmidyllic I'm not sure if we reuse that. `ValidateCredentialStatus` function at the end calls the issuer node to get the credential rev status, and here we are resolving that. Maybe we can talk.